### PR TITLE
Add varnish service definition, fixes drud/ddev#268

### PR DESCRIPTION
**Pull request by stborchert**
Based on the work of [rikwillems](https://github.com/rikwillems) in https://github.com/drud/ddev/issues/268 I created a working example of a varnish service for the current version of DDEV.

### CODE REVIEW

File: docker-compose-services/varnish/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Varnish
+
+This recipe allows you to configure a Varnish reverse proxy for your project.
+
+1. Copy [docker-compose.varnish.yaml](docker-compose.varnish.yaml) into your project's .ddev directory.
+2. Create a directory named _varnish_ in your project's .ddev directory.
+3. Put a file named _default.vcl_ in this directoy.
+4. `ddev start`
+
+For a quick start you can use the put the following contents into _default.vcl_:
```
Comment (rfay) [inline docker-compose-services/varnish/README.md:10]: How about providing this as a file, so people don't miss it? That's what happened at least once in the issue.

**Review by rfay — CHANGES_REQUESTED** [review]
Thanks so much for this. It can have a little more context and split the default.vcl out so people have a starting configuration, but it's a great step forward. Thanks!

File: docker-compose-services/varnish/README.md (hunk 1 above)
Comment (stborchert) [reply docker-compose-services/varnish/README.md]: I've moved the example code to a real file so you can simply copy it. Additionally I've left a comment within the _default.vcl_ with link to a more advanced example.

File: docker-compose-services/varnish/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Varnish
+
+This recipe allows you to configure a Varnish reverse proxy for your project.
```
[the remark this answers is not in the document]
Comment (stborchert) [reply docker-compose-services/varnish/README.md:3]: Tried to explain this (as far as I understand it by myself :) ).

[the remark this answers is not in the document]
Comment (stborchert) [reply docker-compose-services/varnish/README.md:3]: I'm open for and happy about text changes, if that sounds bumpy.

**Review by rfay — APPROVED** [review]

**Pull request merged**